pkg/controller/periodic/periodic.go

```diff
@@ -116,30 +116,32 @@ func (c *Controller) Gather() {
 	allFunctionReports := make(map[string]gather.GathererFunctionReport)
 
 	for _, gatherer := range gatherersToProcess {
-		name := gatherer.GetName()
-		start := time.Now()
+		func() {
+			name := gatherer.GetName()
+			start := time.Now()
 
-		ctx, cancel := context.WithTimeout(context.Background(), c.configurator.Config().Interval/2)
-		defer cancel()
+			ctx, cancel := context.WithTimeout(context.Background(), c.configurator.Config().Interval/2)
+			defer cancel()
```

Comment:
I am wondering if it wouldn't be better to share one context for all the gatherers.

Reply:
I see... That is actually a good point. Let me experiment.

Reply:
Oh, no we can't. 'Cause the timeout.

Reply:
The timeout is 1h by default. I think all the gatherers should be able to finish within this time.

Reply:
Yeh, I know. But it also can be configured, right? Also, would be worth taking the risk?

Reply:
BTW, these changes will directly require changes on tests for periodic. 'Cause, it fails.

Reply:
Sure it can be configured. The risk is always there. I mean if the gathering interval is too short then it will be very likely canceled. I am just saying that I think a 1h time-out for all the gatherers should be enough. Maybe we should keep the discussion in some other task/PR.

Reply:
Yep. I think that we can handle it in some other task, 'cause I'm not sure about it yet.